### DISCUSSION TITLE
In Splade example for MS Marco evaluation why index 8.8M train passages and evaluate wiht 6980 queries from dev  ?

### DISCUSSION THREAD

**Discussion by PrithivirajDamodaran** [castorini/pyserini#1755] [Q&A]
I am referring to the docs page titled "Pyserini: SPLADEv2 on MS MARCO V1 Passage Ranking" 

Please advice.

**Comment by lintool** [comment]
Sorry, not sure if I quite understand the question - this is the standard evaluation methodology for MS MARCO...

**Reply by PrithivirajDamodaran** [reply]
Sorry I didn't know this belongs to the discussion and sorry if I was not clear. 

MS Marco v2.1 has 800k queries spanning to 8.8M million passages in train slice we fine tune models with it and the validation or dev set has 100k queries spanning to 1.1M passages out of 100k queries a standard 6980 queries are designated for literature evaluation. 

I understand this split, but the docs page you show seem to index the train slice passage and evaluate using the 6980 queries from dev / val set. If train slice is used for fine-tuning don't you think it's wrong as the model has seen it ?

**Reply by lintool** [reply]
What page are you referring to? Please cite the URL.

Are you referring to MS MARCO v1 or v2?

**Reply by PrithivirajDamodaran** [reply]
Sure, I am referring to this page.

https://github.com/castorini/pyserini/blob/master/docs/experiments-spladev2.md

Once you open on the header it says Ms Marco v1 but the splade processed tar file (I downloaded the ~9.8gb file) and examined has msmarco v2.1 train slice passages.

The designated 6980 queries are part of msmarco v2.1 validation slice (yes I cross checked it. )

Hence the confusion.

**Reply by lintool** [reply]
I'm still a bit confused. In IR experiments, the corpus remains constant. The train/dev/test splits are only what you use to train the model. But once you train the model, you apply the model to encode the entire corpus.

**Reply by PrithivirajDamodaran** [reply]
I see, so the 8.8M includes all passages form all slices and the queries are 6980 from validation slice. Got it. 

And any reason why the above page says msmarco v1, Possible typo ?

**Reply by lintool** [reply]
No, MS MARCO v1 and v2 are distinct.

**Reply by PrithivirajDamodaran** [reply]
Sorry if I was not clear, so 8.8M is all slices from v2.1, that makes sense and 6980 is a subset from v2.2 dev.

Again sorry for harping on this point is the page heading mistakenly saying v1 ? Because all the data from v2.1

**Reply by lintool** [reply]
Where are you getting "v2.1"? To my knowledge, there is no such thing - there's MS MARCO v1 (8.8M passages) and MS MARCO v2 (138M passages).

**Reply by PrithivirajDamodaran** [reply]
I am referring to this page https://huggingface.co/datasets/ms_marco

Is this not the right source for the dataset ? ![image](https://github.com/castorini/pyserini/assets/7071019/427ea91d-628f-43cb-bc46-739fcedbc46e)

**Reply by lintool** [reply]
This is MS MARCO v1. The hgf dataset was added much later. I've never heard of v1.1 and v2.1 until now.

Refer to the official website: https://microsoft.github.io/msmarco/

**Reply by PrithivirajDamodaran** [reply]
Understood and I have no further questions, thanks for responding.

https://github.com/microsoft/MSMARCO-Passage-Ranking

The download scripts are referring to files with v2.1, so looks like why you are calling v1 is now v2.1 and reflecting in HF as v2.1

J![image](https://github.com/castorini/pyserini/assets/7071019/750dc4da-d2e0-43c0-bfd2-fdec3b741926)ust an fyi I looked at the official website and the repo offices repo:

**Reply by lintool** [reply]
No, I don't think so - MS MARCO is a collection of different datasets. What you are referring to is the QA dataset. For ranking, "msmarcoranking" is that people use.

**Reply by PrithivirajDamodaran** [reply]
If you are so sure then HF should be notified about this version discrepancy. May not be your priority, but I will talk to them and respond here for the benefit of the community.

**Reply by PrithivirajDamodaran** [reply]
Thank you and I would like to take this moment to appreciate all the work with pyserini and related packages - maintaining this is a massive undertaking. Props to you and your team. 

Also I read your whacky weights paper :) very interesting. 

Happy NY.
